cmd/kpod/ps.go

```diff
@@ -33,18 +32,36 @@ type psOptions struct {
 	label   string
 }
 
-type psOutputParams struct {
-	ID         string `json:"id"`
-	Image      string `json:"image"`
-	Command    string `json:"command"`
-	CreatedAt  string `json:"created"`
-	RunningFor string `json:"running"`
-	Status     string `json:"status"`
-	Ports      string `json:"ports"`
-	Size       string `json:"size"`
-	Names      string `json:"names"`
-	Labels     string `json:"labels"`
-	Mounts     string `json:"mounts"`
+type psTemplateParams struct {
+	ID         string
+	Image      string
+	Command    string
+	CreatedAt  string
+	RunningFor string
+	Status     string
+	Ports      string
+	Size       string
+	Names      string
+	Labels     string
+	Mounts     string
+}
+
+// psJSONParams is only used when the JSON format is specified and is better for data processing from JSON.
+// psJSONParams will be populated by data from libkpod.ContainerData,
+// the members of the struct are the sama data types as their sources.
+type psJSONParams struct {
+	ID               string              `json:"id"`
+	Image            string              `json:"image"`
+	Command          string              `json:"command"`
+	CreatedAt        time.Time           `json:"createdAt"`
+	RunningFor       time.Duration       `json:"runningFor"`
+	Status           string              `json:"status"`
+	Ports            map[string]struct{} `json:"ports"`
+	Size             uint                `json:"size"`
+	Names            string              `json:"names"`
+	Labels           fields.Set          `json:"labels"`
+	Mounts           []specs.Mount       `json:"mounts"`
+	ContainerRunning bool                `json:"ctrRunning"`
```

Comment:
No, @baude wanted a boolean variable in the JSON to know where the container is running or not, so I added it in psJSONParams. psTemplateParams doesn't need this.